store/app/routers/stripe.py

```diff
@@ -40,6 +40,65 @@ async def create_payment_intent(request: Request) -> Dict[str, Any]:
         return {"error": str(e)}
 
 
+class CancelReason(BaseModel):
+    reason: str
+    details: str
+
+
+class CreateRefundsRequest(BaseModel):
+    payment_intent_id: str
+    cancel_reason: CancelReason
+    amount: int
+
+
+@stripe_router.put("/refunds/{order_id}", response_model=Order)
+async def refund_payment_intent(
+    order_id: str,
+    refund_request: CreateRefundsRequest,
+    user: User = Depends(get_session_user_with_read_permission),
+    crud: Crud = Depends(),
+) -> Order:
+    async with crud:
+        try:
+            amount = refund_request.amount
+            payment_intent_id = refund_request.payment_intent_id
+            customer_reason = (
+                refund_request.cancel_reason.details
+                if (refund_request.cancel_reason.reason == "Other" and refund_request.cancel_reason.details)
+                else refund_request.cancel_reason.reason
+            )
+
+            # Create a Refund for payment_intent_id with the order amount
+            refund = stripe.Refund.create(
+                payment_intent=payment_intent_id,
+                amount=amount,
+                reason="requested_by_customer",
+                metadata={"customer_reason": customer_reason},
+            )
+            logger.info(f"Refund created: {refund.id}")
+
+            # Make sure order exists
+            order = await crud.get_order(order_id)
+            if order is None or order.user_id != user.id:
+                raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Order not found")
+            logger.info(f"Found order id: {order.id}")
```

Comment:
very nit: new line here
```suggestion
            logger.info(f"Found order id: {order.id}")

```